### PR TITLE
Make `And` a leaf bloq

### LINKED ISSUE #1346: Make `And` a leaf bloq
Specific changes that are needed to do this part of #873 :

 - Any unit test that relies on the "sigma" return value of `call_graph` will change. This is brittle to begin with. Need to port to `QECGatesCost` 
 - Cirq simulation needs the cirq-style decomposition unless we override `_kraus_` with custom kraus operators. 
 - Shim over the `_t_complexity_` values for the time being.

### PR DESCRIPTION
**Pull request by anurudhp**
fixes #1346 

also removes the only use of big O in call graph #1368

- first commit sets `FLAG_AND_IS_LEAF=True` and fixes all tests
- second one removes it altogether and cleans up the `And` bloq code
- fix `t_counts_from_sigma` to explicitly count `And`s

### CODE REVIEW

**Comment by anurudhp** [comment]
@mpharrigan @fdmalone ptal!

File: qualtran/bloqs/chemistry/df/double_factorization.py [hunk 1]
```diff
@@ -280,10 +279,10 @@ def build_call_graph(self, ssa: 'SympySymbolAllocator') -> 'BloqCountDictT':
             in_prep_dag: 1,  # in_prep_l^dag
             rot: 1,  # rotate into system basis  listing 4 pg 54
             # apply CCZ first then CCCZ, the cost is 1 + 2 Toffolis (step 4e, and 7)
-            Toffoli(): 1,
+            ApplyControlledZs(cvs=(1, 1), bitsize=5): 1,
```
Comment (mpharrigan) [inline qualtran/bloqs/chemistry/df/double_factorization.py:282]: was this incorrect before? I see the change makes it match the comment

**Review by mpharrigan — APPROVED** [review]
excellent. very nice

File: qualtran/bloqs/chemistry/df/double_factorization.py (hunk 1 above)
Comment (anurudhp) [reply qualtran/bloqs/chemistry/df/double_factorization.py:282]: The previous test was only checking the `sigma` for the call graph. I updated it to check for equivalent bloq counts, which flagged this inconsistency.

**Pull request merged**